### PR TITLE
fix #120, address sanitizer issue in fm_dispatch unit test

### LINKED ISSUE #120: Address Sanitizer issue in coverage-fm-fm_dispatch
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/FM/blob/main/CONTRIBUTING.md).
* [x] I performed a cursory search to see if the bug report is relevant, not redundant, nor in conflict with other tickets.

**Describe the bug**

Running with F sanitizer triggers a runner time error. 

**To Reproduce**
Add the following option:
add_compile_options(
-fsanitize=address
-fsanitize=undefined
-fsanitize-recover=all
-g
)
add_link_options(
-fsanitize=address
-fsanitize=undefined
-fsanitize-recover=all
)

Make SIMULATION=native ENABLE_UNIT_TESTS=true prep

make install

Run Test

**Expected behavior**

**Code snips**
If applicable, add references to the software.

**System observed on:**
 - Hardware
 - OS: [e.g. Linux 4.4]
 - Versions [e.g. cFE 6.6, OSAL 4.2, PSP 1.3 for mcp750, any related apps]

**Additional context**

**Reporter Info**
Full name and company/organization if applicable

Anh Van, GSFC

### PR DESCRIPTION
**Pull request by avan989**
fix #120, fix passing in null as parameter in unit test

**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/FM/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate [Contributor License Agreement](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla) to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
A clear and concise description of what the contribution is.
- Include explicitly what issue it addresses [e.g. Fixes #X]
fixes #120, fix run time error with address sanitizer. 

**Testing performed**
Steps taken to test the contribution:
Add the following option:
add_compile_options(
-fsanitize=address
-fsanitize=undefined
-fsanitize-recover=all
-g
)
add_link_options(
-fsanitize=address
-fsanitize=undefined
-fsanitize-recover=all
)

Make SIMULATION=native ENABLE_UNIT_TESTS=true prep

make install

Run Test

**Expected behavior changes**
A clear and concise description of how this contribution will change behavior and level of impact.
 - API Change: xxx (if applicable)
 - Behavior Change: xxx (if applicable)
 - Or no impact to behavior

**System(s) tested on**
 - Hardware: [e.g. PC, SP0, MCP750]
 - OS: [e.g. Ubuntu 18.04, RTEMS 4.11, VxWorks 6.9]
 - Versions: [e.g. cFE 6.6, OSAL 4.2, PSP 1.3 for mcp750, any related apps or tools]

**Additional context**
Add any other context about the contribution here.

**Third party code**
If included, identify any third party code and provide text file of license

**Contributor Info - All information REQUIRED for consideration of pull request**
Full name and company/organization/center of all contributors ("Personal" if individual work)
 - Note CLA's apply to software contributions.
Anh Van, GSFC

### CODE REVIEW

**Comment by dzbaker** [comment]
CCB 20 June 2024: Approved pending review by @jphickey.

**Comment by avan989** [comment]
updated to amend commit message. @dzbaker can you rerun the workflow?

**Comment by dzbaker** [comment]
CCB 8 August 2024: Approving at recommendation of @jphickey.

**Review by jphickey — APPROVED** [review]

**Pull request merged**
